waiter/src/waiter/headers.clj

```diff
@@ -113,13 +113,12 @@
 (defn assoc-auth-headers
   "`assoc`s the x-waiter-auth-principal and x-waiter-authenticated-principal headers if the
    username and principal are non-nil, respectively."
-  ([headers principal]
-   (let [username (first (str/split principal #"@" 2))]
-     (assoc-auth-headers headers username principal)))
-  ([headers username principal]
-   (cond-> headers
-     username (assoc "x-waiter-auth-principal" username)
-     principal (assoc "x-waiter-authenticated-principal" principal))))
+  ([headers principal {:keys [access-token]}]
+   (let [username (utils/principal->username principal)]
+     (cond-> headers
+       username (assoc "x-waiter-auth-principal" username)
+       principal (assoc "x-waiter-authenticated-principal" principal)
+       access-token (assoc "x-waiter-jwt" access-token)))))
```

Comment:
We should only add `access-token` if the method is `:jwt`.  In theory other auth methods might add `access-token` and it wouldn't be correct to add `x-waiter-jwt`.

Reply:
Changed the key to `jwt-access-token`. When we support `OIDC`, the method will be `oidc` and will have a token to send to the backend with the same key.